Crashes a couple of seconds after finishing

Crash log 
[crash-2023-01-19_17.50.16-client.txt](https://github.com/TimStewartJ/TheMightyArchitectury/files/10456502/crash-2023-01-19_17.50.16-client.txt)

I'm having a similar issue.  I tried with the new zip you posted, and I'm getting this error in the log a bunch of times:
[13:13:35] [Render thread/ERROR]: 
 CCL has caught an exception whilst rendering a block
  BlockPos:      x:-6360, y:69, z:2034
  Block Class:   class net.minecraft.world.level.block.RotatedPillarBlock
  Registry Name: minecraft:spruce_wood
  State:         Block{minecraft:spruce_wood}[axis=y]
 Tile at position
  Tile Class:    null
  Tile Id:       null
  Tile NBT:      null
This functionality can be disabled in the CCL config file.
You can also turn off player messages in the CCL config file.

net.minecraft.ReportedException: Tesselating block model
	at net.minecraft.client.renderer.block.BlockRenderDispatcher.renderBatched(BlockRenderDispatcher.java:78) ~[client-1.19.2-20220805.130853-srg.jar%23613!/:?]
	at codechicken.lib.render.block.CCBlockRendererDispatcher.renderBatched(CCBlockRendererDispatcher.java:71) ~[CodeChickenLib-1.19.2-4.3.1.481-universal.jar%23415!/:4.3.1.481]
	at net.minecraft.client.renderer.block.BlockRenderDispatcher.renderBatched(BlockRenderDispatcher.java:65) ~[client-1.19.2-20220805.130853-srg.jar%23613!/:?]
	at net.minecraft.client.renderer.block.BlockRenderDispatcher.m_234355_(BlockRenderDispatcher.java:62) ~[client-1.19.2-20220805.130853-srg.jar%23613!/:?]
	at com.timmie.mightyarchitect.control.SchematicRenderer.lambda$redraw$0(SchematicRenderer.java:113) ~[mightyarchitect-0.6.1-FORGE.jar%23519!/:?]
	at java.util.Iterator.forEachRemaining(Iterator.java:133) ~[?:?]
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1845) ~[?:?]
	at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:762) ~[?:?]
	at com.timmie.mightyarchitect.control.SchematicRenderer.redraw(SchematicRenderer.java:93) ~[mightyarchitect-0.6.1-FORGE.jar%23519!/:?]
	at com.timmie.mightyarchitect.control.SchematicRenderer.tick(SchematicRenderer.java:60) ~[mightyarchitect-0.6.1-FORGE.jar%23519!/:?]
	at com.timmie.mightyarchitect.MightyClient.onTick(MightyClient.java:52) ~[mightyarchitect-0.6.1-FORGE.jar%23519!/:?]
	at java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:732) ~[?:?]
	at dev.architectury.event.EventFactory.invokeMethod(EventFactory.java:53) ~[architectury-6.5.77-forge.jar%23391!/:?]
	at dev.architectury.event.EventFactory$1.handleInvocation(EventFactory.java:62) ~[architectury-6.5.77-forge.jar%23391!/:?]
	at com.google.common.reflect.AbstractInvocationHandler.invoke(AbstractInvocationHandler.java:89) ~[guava-31.0.1-jre.jar%23115!/:?]
	at jdk.proxy3.$Proxy120.tick(Unknown Source) ~[?:?]
	at dev.architectury.event.forge.EventHandlerImplClient.event(EventHandlerImplClient.java:57) ~[architectury-6.5.77-forge.jar%23391!/:?]
	at dev.architectury.event.forge.__EventHandlerImplClient_event_ClientTickEvent.invoke(.dynamic) ~[architectury-6.5.77-forge.jar%23391!/:?]
	at net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:73) ~[eventbus-6.0.3.jar%2379!/:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:315) ~[eventbus-6.0.3.jar%2379!/:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:296) ~[eventbus-6.0.3.jar%2379!/:?]
	at net.minecraftforge.event.ForgeEventFactory.onPostClientTick(ForgeEventFactory.java:830) ~[forge-1.19.2-43.2.6-universal.jar%23618!/:?]
	at net.minecraft.client.Minecraft.m_91398_(Minecraft.java:1852) ~[client-1.19.2-20220805.130853-srg.jar%23613!/:?]
	at net.minecraft.client.Minecraft.m_91383_(Minecraft.java:1078) ~[client-1.19.2-20220805.130853-srg.jar%23613!/:?]
	at net.minecraft.client.Minecraft.m_91374_(Minecraft.java:700) ~[client-1.19.2-20220805.130853-srg.jar%23613!/:?]
	at net.minecraft.client.main.Main.m_239872_(Main.java:212) ~[client-1.19.2-20220805.130853-srg.jar%23613!/:?]
	at net.minecraft.client.main.Main.main(Main.java:51) ~[client-1.19.2-20220805.130853-srg.jar%23613!/:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at net.minecraftforge.fml.loading.targets.CommonClientLaunchHandler.lambda$launchService$0(CommonClientLaunchHandler.java:27) ~[fmlloader-1.19.2-43.2.6.jar%2395!/:?]
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:30) [modlauncher-10.0.8.jar%2382!/:?]
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) [modlauncher-10.0.8.jar%2382!/:?]
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) [modlauncher-10.0.8.jar%2382!/:?]
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:106) [modlauncher-10.0.8.jar%2382!/:?]
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:77) [modlauncher-10.0.8.jar%2382!/:?]
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) [modlauncher-10.0.8.jar%2382!/:?]
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) [modlauncher-10.0.8.jar%2382!/:?]
	at cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:141) [bootstraplauncher-1.1.2.jar:?]
Caused by: java.lang.IndexOutOfBoundsException
	at java.nio.Buffer.checkIndex(Buffer.java:749) ~[?:?]
	at java.nio.DirectByteBuffer.putShort(DirectByteBuffer.java:625) ~[?:?]
	at com.mojang.blaze3d.vertex.BufferBuilder.m_5586_(BufferBuilder.java:279) ~[client-1.19.2-20220805.130853-srg.jar%23613!/:?]
	at com.mojang.blaze3d.vertex.BufferBuilder.handler$zoh000$iris$beforeNext(BufferBuilder.java:2205) ~[client-1.19.2-20220805.130853-srg.jar%23613!/:?]
	at com.mojang.blaze3d.vertex.BufferBuilder.m_5752_(BufferBuilder.java) ~[client-1.19.2-20220805.130853-srg.jar%23613!/:?]
	at com.mojang.blaze3d.vertex.VertexConsumer.m_5954_(VertexConsumer.java:39) ~[client-1.19.2-20220805.130853-srg.jar%23613!/:?]
	at com.mojang.blaze3d.vertex.BufferBuilder.m_5954_(BufferBuilder.java:2605) ~[client-1.19.2-20220805.130853-srg.jar%23613!/:?]
	at com.mojang.blaze3d.vertex.VertexConsumer.putBulkData(VertexConsumer.java:115) ~[client-1.19.2-20220805.130853-srg.jar%23613!/:?]
	at com.mojang.blaze3d.vertex.VertexConsumer.m_85995_(VertexConsumer.java:67) ~[client-1.19.2-20220805.130853-srg.jar%23613!/:?]
	at com.mojang.blaze3d.vertex.BufferBuilder.m_85995_(BufferBuilder.java:1576) ~[client-1.19.2-20220805.130853-srg.jar%23613!/:?]
	at net.minecraft.client.renderer.block.ModelBlockRenderer.m_111023_(ModelBlockRenderer.java:154) ~[client-1.19.2-20220805.130853-srg.jar%23613!/:?]
	at net.minecraft.client.renderer.block.ModelBlockRenderer.m_111012_(ModelBlockRenderer.java:134) ~[client-1.19.2-20220805.130853-srg.jar%23613!/:?]
	at net.minecraft.client.renderer.block.ModelBlockRenderer.tesselateWithAO(ModelBlockRenderer.java:88) ~[client-1.19.2-20220805.130853-srg.jar%23613!/:?]
	at net.minecraftforge.client.model.lighting.ForgeModelBlockRenderer.tesselateWithAO(ForgeModelBlockRenderer.java:65) ~[forge-1.19.2-43.2.6-universal.jar%23618!/:?]
	at net.minecraft.client.renderer.block.ModelBlockRenderer.tesselateBlock(ModelBlockRenderer.java:58) ~[client-1.19.2-20220805.130853-srg.jar%23613!/:?]
	at net.minecraft.client.renderer.block.BlockRenderDispatcher.renderBatched(BlockRenderDispatcher.java:71) ~[client-1.19.2-20220805.130853-srg.jar%23613!/:?]
	... 39 more

--- a/common/src/main/java/com/timmie/mightyarchitect/control/compose/planner/LayerStyleTool.java
+++ b/common/src/main/java/com/timmie/mightyarchitect/control/compose/planner/LayerStyleTool.java
@@ -51,7 +51,7 @@ public class LayerStyleTool extends WallDecorationToolBase {
 		List<DesignLayer> layers = model.getTheme().getRoomLayers();
 
 		int index = ArrayUtils.indexOf(Minecraft.getInstance().options.keyHotbarSlots, mapping.get());
-		if (index > layers.size())
+		if (index >= layers.size())
 			return;
 
 		DesignLayer newLayer = layers.get(index);

